src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -340,6 +327,39 @@ private function parseDefinition($id, $service, $file)
     }
 
     /**
+     * Parses a callable.
+     *
+     * @param string|array $callable  A callable
+     * @param string       $parameter A parameter (e.g. 'factory' or 'configurator')
+     * @param string       $id        A service identifier
+     * @param string       $file      A parsed file
+     *
+     * @throws InvalidArgumentException When errors are occuried
+     *
+     * @return string|array A parsed callable
+     */
+    private function parseCallable($callable, $parameter, $id, $file)
+    {
+        if (is_string($callable)) {
+            if (false !== strpos($callable, ':') && false === strpos($callable, '::')) {
+                $parts = explode(':', $callable);
+
+                return array($this->resolveServices('@'.$parts[0]), $parts[1]);
+            }
+
+            return $callable;
+        } elseif (is_array($callable)) {
```

Comment:
@hhamon, You're right, with standalone `if` code is more readable. I've changed `elseif` to `if`. Thanks!